index.js

```diff
@@ -0,0 +1,107 @@
+const express = require('express');
+const { Client } = require('pg');
+const client = new Client({
+    user: 'postgres',
+    host: 'localhost',
+    database: 'movies_database',
+    password: 'yojikyo',
+    port: 5432,
+});
+client.connect();
+const sendResponceByQuery = (query, paramsNames, onAcceptQuery) => {
+    return (request, response) => {
+        query.values = paramsNames.map(name => request.params[name]);
+        client.query(
+            query,
+            (error, result) => {
+            if (error) {
+                response.sendStatus(500);
+                console.log(error);
+            } else {
+                onAcceptQuery(response, result);
+            }
+        });
+    };
+};
+
+const app = express();
+
+const queries = require('./queries');
+
+app.get('/movies', sendResponceByQuery(
+    queries.getAllMovies,
+    [],
+    (response, queryResult) => {
+        response.json(queryResult.rows);
+    }
+));
+
+app.get('/movies/:movieId(\\d+)', sendResponceByQuery(
+    queries.getMovieById,
+    ['movieId'],
+    (response, queryResult) => {
+        if (queryResult.rows.length === 0) {
+            response.sendStatus(404);
+        } else {
+            response.json(queryResult.rows[0]);
+        }
+    }
+));
+
+app.get('/movies/titles', sendResponceByQuery(
+    queries.getAllTitles,
+    [],
+    (response, queryResult) => {
+        if (queryResult.rows.length === 0) {
+            response.sendStatus(204);
+        } else {
+            response.send(queryResult.rows.reduce((resultText, { title }) => {
+                return resultText + title + '<br>';
+            }, ''));
+        }
+    }
+));
+
+app.get('/movies/titlesByYear/:year(\\d{4})', sendResponceByQuery(
+    queries.getTitlesByYear,
+    ['year'],
+    (response, queryResult) => {
+        if (queryResult.rows.length === 0) {
+            response.sendStatus(204);
+        } else {
+            response.send(queryResult.rows.reduce((resultText, { title }, index, inputArray) => {
+                return resultText + title + ';' + (index === inputArray.length ? '' : '<br>');
```

Comment:
use template string
the best solution here will be `join`, reduce is overkill
index always < array.length